snippet_parser/core_test.py

```diff
@@ -146,5 +146,63 @@ def test_no_wikicode_in_section_titles(self):
             "== Section title ''with Wikicode'' == \n\nIrrelevant content{{cn}}")
         self.assertEqual(section, 'Section title with Wikicode')
 
+    def _do_extract_sentences(self, wikitext, sentences):
+        for sentence in sentences:
+            start = wikitext.find(sentence)
+            if start < 0: continue
+            marked = core._SENTENCE_MARKER_MARKUP.format(sent = sentence)
+            wikitext = wikitext[:start] + marked + wikitext[start+len(sentence):]
+        self._set_wikipedia_parse_response(wikitext)
+        ret = self._sp.extract_from_sentences(wikitext, sentences)
+        call_args = None
+        if self._wp.parse.call_count:
+            call_args = self._wp.parse.call_args[0][0]
+        ret = ret[0] if ret else []
+        return call_args, ret
+
+    def test_sentence_to_snippet(self):
+        sentences = ['A sentence.']
+        wikitext = '<p>Stuff. A sentence. More stuff.</p>'
```

Comment:
I think you don't want the `<p>` tags, here and in the other tags, right? If this is meant to be Wikitext, we shouldn't be putting in HTML tags.